crates/store/re_server/src/store/dataset.rs

```diff
@@ -44,6 +47,32 @@ impl Dataset {
         &self.name
     }
 
+    pub fn partition(&self, partition_id: &PartitionId) -> Result<&Partition, Error> {
+        self.partitions
+            .get(partition_id)
+            .ok_or_else(|| Error::PartitionIdNotFound(partition_id.clone(), self.id))
+    }
+
+    pub fn partitions_from_ids<'a>(
+        &'a self,
+        partition_ids: &'a [PartitionId],
+    ) -> Result<impl Iterator<Item = (&'a PartitionId, &'a Partition)>, Error> {
+        if partition_ids.is_empty() {
+            Ok(Either::Left(self.partitions.iter()))
```

Comment:
what? why are we returning all of them if we asked for none?! That's not documented and if it was I'd be complaining about it :P

Reply:
That is not new. A common pattern in our protos, and a rare instance of something that _is_ actually documented:

https://github.com/rerun-io/rerun/blob/20ddeddbaf2d030b15d680738a4f40763c54e85b/crates/store/re_protos/proto/rerun/v1alpha1/cloud.proto#L427-L430

It's also tested.

Adding a docstring to clarify.